undo call{T}(::Type{Node{T}}, x) deprecation

sjkelly opened the pull request:
This undoes the deprecation introduced in #65 for `call{T}(::Type{Node{T}}, x)`. AFAIK it wasn't an issue with syntax as much as the difficulty of finding the right `Base.call` (arggghh type invariance!).

sjkelly committed:
undo call{T}(::Type{Node{T}}, x) deprecation

shashi commented:
Is there a way to keep `Signal(T, value)` working?

I think this is consistent with the Julia convention, e.g. `Array(T, 10)` etc.

sjkelly commented:
Yes, I can do that. Since 0.4 `Array{T}(10)` has been supported with call, so it is helpful to have both I guess.

sjkelly commented:
I lost some motivation rebasing for the `Signal` rename, so I just removed the deprecation and added constructor tests. This makes the patch much smaller.

sjkelly commented:
Bump.
